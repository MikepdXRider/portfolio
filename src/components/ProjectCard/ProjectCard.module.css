:root {
  /* alternatives: hsl(225,7%,22%) | rgb(53,55,61)*/
  --darkslategrey: #35373d;
  /* alternatives: hsl(0,100%,100%) | rgb(255,255,255)*/
  --white: #ffffff;
  /* alternatives: hsl(0,0%,100%) | rgb(0,0,0)*/
  --black: #000000;
  /* alternatives: hsl(90,58%,56%) | rgb(143,209,79)*/
  --accentgreen: #8fd14f;
}

/* The flip card container - set the width and height to whatever you want. We have added the border property to demonstrate that the flip itself goes out of the box on hover (remove perspective if you don't want the 3D effect */
.flipCard {
  background-color: transparent;
  height: 260px;
  min-width: 240px;
  perspective: 1000px; /* Remove this if you don't want the 3D effect */
}

/* This container is needed to position the front and back side */
.flipCardInner {
  border-radius: 10px;
  position: relative;
  width: 100%;
  height: 100%;
  text-align: center;
  transition: transform 0.8s;
  transform-style: preserve-3d;
  background-color: var(--darkslategrey);
  color: var(--white);
  border: var(--darkslategrey) 3px solid;
}

/* Do an horizontal flip when you move the mouse over the flip box container */
.flipCard:hover .flipCardInner {
  transform: rotateY(180deg);
}

/* Position the front and back side */
.flipCardFront,
.flipCardBack {
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-backface-visibility: hidden; /* Safari */
  backface-visibility: hidden;
  font-size: 20px;
  padding: 5px;
  box-shadow: 5px 5px 10px 5px var(--black);
  border-radius: 10px;
}

/* TODO: Style the front side (fallback if image is missing) */

/* Style the back side */
.flipCardBack {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  transform: rotateY(180deg);
}

/* Flip Card front side content */
.title {
  color: var(--accentgreen);
  font-size: 24px;
}

.techList {
  margin-top: 5px;
}

.techListItem {
  list-style-type: none;
}

/* Flip Card back side content */
.linkContainer {
  width: 100%;
  display: flex;
  justify-content: space-around;
}

.link {
  text-decoration: none;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 45%;
  height: 30px;
  font-size: 20px;
  border-radius: 5px;
  background-color: var(--accentgreen);
  color: var(--darkslategrey);
  border: 1px solid var(--black);
}
