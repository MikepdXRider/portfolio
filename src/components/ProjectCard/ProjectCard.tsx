import React from 'react';

interface ProjectCardProps {
  title: string;
  techStack: string[];
  description: string;
  githubUrl: string;
  livesiteUrl: string;
}

export default function ProjectCard({
  title,
  techStack,
  description,
  githubUrl,
  livesiteUrl,
}: ProjectCardProps) {
  return (
    <article className={styles.flipCard}>
      <div className={styles.flipCardInner}>
        <div className={styles.flipCardFront}>
          <h2>{title}</h2>
          <ul>
            {techStack.map((techItem) => (
              <li key={techItem}>{techItem}</li>
            ))}
          </ul>
        </div>
        <div className={styles.flipCardBack}>
          <h3>{description}</h3>
          <a href={githubUrl}>Github</a>
          <a href={livesiteUrl}>Livesite</a>
        </div>
      </div>
    </article>
  );
}
