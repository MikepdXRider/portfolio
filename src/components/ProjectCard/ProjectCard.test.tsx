import { getByRole, render, screen } from '@testing-library/react';
import ProjectCard from './ProjectCard';

describe('ProjectCard component', () => {
  const mockProject = {
    title: 'Mock-project-title',
    techStack: ['Mock-tech-one', 'Mock-tech-two', 'Mock-tech-three'],
    description: 'Mock project description',
    githubUrl: 'https://mock-github.com/',
    livesiteUrl: 'https://mock-livesite.com/',
  };

  it('matches the existing snapshot', () => {
    const { container } = render(<ProjectCard {...mockProject} />);

    expect(container).toMatchSnapshot();
  });

  it('renders component with all passed information', () => {
    render(<ProjectCard {...mockProject} />);

    // front of card
    screen.getByText(/mock-project-title/i);
    screen.getByText(/mock-tech-one/i);
    screen.getByText(/mock-tech-two/i);
    screen.getByText(/mock-tech-three/i);

    // back of card
    screen.getByText(/mock project description/i);
    screen.getByLabelText(/mock-project-title livesite link/i);
    screen.getByLabelText(/mock-project-title github link/i);
  });
});
