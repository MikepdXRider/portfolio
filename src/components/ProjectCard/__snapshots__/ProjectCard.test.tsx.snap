// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard component matches the existing snapshot 1`] = `
<div>
  <article
    class="flipCard"
  >
    <div
      class="flipCardInner"
    >
      <div
        class="flipCardFront"
      >
        <h2
          class="title"
        >
          Mock-project-title
        </h2>
        <ul
          class="techList"
        >
          <li
            class="techListItem"
          >
            Mock-tech-one
          </li>
          <li
            class="techListItem"
          >
            Mock-tech-two
          </li>
          <li
            class="techListItem"
          >
            Mock-tech-three
          </li>
        </ul>
      </div>
      <div
        class="flipCardBack"
      >
        <p>
          Mock project description
        </p>
        <div
          class="linkContainer"
        >
          <a
            aria-label="Mock-project-title github link. Opens a new tab."
            class="link"
            href="https://mock-github.com/"
            rel="noreferrer"
            target="_blank"
          >
            Github
          </a>
          <a
            aria-label="Mock-project-title livesite link. Opens a new tab."
            class="link"
            href="https://mock-livesite.com/"
            rel="noreferrer"
            target="_blank"
          >
            Livesite
          </a>
        </div>
      </div>
    </div>
  </article>
</div>
`;
